src/modules/emotes.js

```diff
@@ -333,176 +293,9 @@ function Emote(details) {
 			throw new Error('Invalid channel');
 		}
 
-		// Only save the channel to storage if it's dynamic.
-		if (theChannel !== '~global' && theChannel !== 'turbo' && theChannel !== 'twitch_prime') {
-			storage.channelNames.set(this.getText(), theChannel);
-		}
 		channel.name = theChannel;
 	};
 
-	/**
-	 * Gets the emote channel's badge image URL.
-	 * @return {string|null} The URL of the badge image for the emote's channel or `null` if it doesn't have a channel.
-	 */
-	this.getChannelBadge = function () {
```

Comment:
In place of this, what do you think about using the profile image as opposed to badges considering badges aren't really seen except when on the channel's own chat whereas the profile image is used all over the place, e.g. some would be:

Mine: ![](https://static-cdn.jtvnw.net/jtv_user_pictures/cletusc-profile_image-e0c7fb2ad09fd5b0-70x70.png), DansGaming: ![](https://static-cdn.jtvnw.net/jtv_user_pictures/dansgaming-profile_image-76e4a4ab9388bc9c-70x70.png)... or others: ![](https://i.imgur.com/RknXEaq.png)

Reply:
To add, this would be dead simple to add--it's already part of the data on the user API for display name and ID lookups.

Reply:
You’d have to make sure it’s in the existing API calls.. it’s expensive for us to make as many api calls as you were making

Reply:
Although I agree that there were a bunch of individual calls, the badge images (along with display names) were cached in localStorage so after the first hit, it didn't hit for quite some time. With these changes, you removed both badges and display names. Badges (or alternatively like I mentioned above, the profile image), along with display names, should still be part of the menu.

The v5 API (and even the new API) allows for multiple usernames lookups in a single call, meaning display names and profile images would incur a single API call. Once you cache that, the API doesn't even get hit unless there's a new name or the user clears localStorage.

Reply:
It’s not just the act of calling an API, localstorage is slow to read from, and you individually store each. Localstorage reads and writes to disk.. I didn’t feel like refactoring it so I removed it. In the grand scheme it doesn’t matter if we have it or not

Reply:
The storage API I made doesn't actually read from localStorage every time either, [it actually caches into memory](https://github.com/cletusc/storage-wrapper/blob/master/index.js#L132-L135) meaning it only reads/writes once per page load per key. This could probably be further reduced to just a few keys instead (as the storage library already supports object storage out of the box without messing with JSON).

This will definitely go back in one way or another, but I'll review how best to reduce API calls and subsequent localStorage calls.